### PR TITLE
feat: introduce timeout

### PR DESCRIPTION
**Pull request by strophy**
This PR is based on #22 and also adds a function to specify the timeout after creating the RPC client. I need help understanding how callbacks are supposed to work inside promises, and how to test this function properly.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ function RpcClient(opts) {
   this.port = opts.port || 9998;
   this.user = opts.user || 'user';
   this.pass = opts.pass || 'pass';
+  this.timeout = opts.timeout || null;
```
Comment (nmarley) [inline lib/index.js:13]: Should this default to 0 (disabled) or some other number instead? I'm just wondering what happens below if this is set to `null`, would it trigger some exception?

Comment (shumkov) [reply lib/index.js:13]: Not sure, that `0` or `null` is the default for https://nodejs.org/api/https.html#httpsrequestoptions-callback. I would probably just keep `undefined`.

File: lib/index.js [hunk 2]
```diff
@@ -74,6 +75,7 @@ function innerRpc(request, callback) {
     port: self.port,
     rejectUnauthorized: self.rejectUnauthorized,
     agent: self.disableAgent ? false : undefined,
+    timeout: self.timeout,
```
Comment (shumkov) [inline lib/index.js:78]: I would add it only in case it's defined. We don't know what the default value is.

**Review by shumkov — CHANGES_REQUESTED** [review]

File: lib/index.js [hunk 3]
```diff
@@ -74,6 +75,7 @@ function innerRpc(request, callback) {
     port: self.port,
     rejectUnauthorized: self.rejectUnauthorized,
     agent: self.disableAgent ? false : undefined,
+    timeout: self.timeout ? self.timeout : undefined,
```
Comment (shumkov) [inline lib/index.js:78]:
```suggestion
};

if (self.timeout) {
  options.timeout = self.timeout;
```

**Review by nmarley — APPROVED** [review]
utACK

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**